containerd-shim-v2/service.go

```diff
@@ -818,6 +910,9 @@ func (s *service) Stats(ctx context.Context, r *taskAPI.StatsRequest) (_ *taskAP
 
 // Update a running container
 func (s *service) Update(ctx context.Context, r *taskAPI.UpdateTaskRequest) (_ *ptypes.Empty, err error) {
+	span := trace(s.ctx, "Update")
```

Comment:
@jodh-intel  I'm curious for your input here. 

AFAIU, for shimv2, we'll get a context for the life of the shim itself, which is what we store in s.ctx (I think it may be wise to rename this to be shimCtx to make it more clear). In this case, when we're called, we have a context associated with the given request, which has a 'shorter' scope, ctx.

1. Should we use ctx in the 'short lived' maintenance requests rather than s.ctx? 
2. Should we 'plumb' ctx through to the actual virtcontainers impl here? 

Basically, I am starting to think that we should do this per 'request' into the VC API rather than having a sandbox level ctx that is stored in the struct. The struct makes sense maybe when we're just doing traces, but I think we should be able to also handle contexts per request from shim-v2, in case of timeouts, cancellations, etc. First step would at least be to plumb this through?

Reply:
/cc @lifupan for input as well